libpromises/locks.c

```diff
@@ -554,6 +554,34 @@ static void PromiseTypeString(char *dst, size_t dst_size, const Promise *pp)
     }
 }
 
+// returns true if pid points to a cfengine process
+static bool IsCfengineProcess(pid_t pid)
+{
+    char procfile[PATH_MAX];
+    char cmd[PATH_MAX]; // we don't need the full ARG_MAX
+    snprintf(procfile, PATH_MAX, "/proc/%d/cmdline", pid);
```

Comment:
`pid_t` doesn't have to be `int` and this can easily fail to compile in such a case because of the format (`%d`) and type (`pid_t`) mismatch. We usually use a trick with `uintmax_t` for things like this.